core/src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala

```diff
@@ -195,12 +196,18 @@ class CassandraSnapshotStore(cfg: Config) extends SnapshotStore
   }
 
   override def deleteAsync(persistenceId: String, criteria: SnapshotSelectionCriteria): Future[Unit] = {
-    preparedSelectSnapshotMetadata.flatMap { prepStmt =>
-      metadata(prepStmt, persistenceId, criteria, limit = None).flatMap { mds =>
-        val boundStatements = mds.map(md => preparedDeleteSnapshot.map(_.bind(md.persistenceId, md.sequenceNr: JLong)))
-        Future.sequence(boundStatements).flatMap { stmts =>
-          executeBatch(batch => stmts.foreach(batch.add))
+    preparedSelectSnapshotMetadata.flatMap { snapshotMetaPs =>
+      // this meta query gets slower than slower if snaphosts are deleted
+      metadata(snapshotMetaPs, persistenceId, criteria, limit = None).flatMap { mds: immutable.Seq[SnapshotMetadata] =>
+        val boundStatements: immutable.Seq[Future[BoundStatement]] = mds.map(md => preparedDeleteSnapshot.map(_.bind(md.persistenceId, md.sequenceNr: JLong)))
```

Comment:
you like explicit types today? leftover from making compiler happy?

Reply:
Not sure what i was doing there :)